tools/magfit_WMM.py

```diff
@@ -245,21 +241,32 @@ def remove_offsets(MAG, BAT, c):
     MAG.MagZ = int(field.z)
     return True
 
+def param_name(short_name, index):
+    global new_param_format
+    if new_param_format:
+        return "COMPASS%s_%s" % (index, short_name)
+    if index == 1:
+        return "COMPASS_%s" % short_name
+    return "COMPASS_%s%s" % (short_name, index)
+
 def magfit(logfile):
     '''find best magnetometer offset fit to a log file'''
 
     print("Processing log %s" % logfile)
     mlog = mavutil.mavlink_connection(logfile)
 
-    global earth_field, declination
+    global earth_field, declination, new_param_format
 
     global data
     data = []
 
     ATT = None
     BAT = None
 
-    mag_msg = 'MAG%s' % mag_idx
+    if args.mag == 1:
```

Comment:
this fails for old logs with MAG, MAG2, MAG3

Reply:
still works, tested on a 4.0 log MAG and MAG2